poc_iot_injector/src/server.rs

```diff
@@ -111,20 +117,25 @@ async fn submit_txns(
 
     store
         .source_unordered(LOADER_WORKERS, stream::iter(file_list).map(Ok).boxed())
-        .for_each_concurrent(STORE_WORKERS, |msg| {
+        .try_for_each_concurrent(STORE_WORKERS, |msg| {
             let mut shared_txn_service = txn_service.clone();
             let shared_key = keypair.clone();
             async move {
-                match msg {
-                    Ok(m) => {
-                        handle_txn_submission(m, shared_key, &mut shared_txn_service, before_ts)
-                            .await
+                if let Some(txn) =
+                    handle_txn_submission(msg, shared_key, &mut shared_txn_service, before_ts).await
+                {
+                    // NOTE: Can the txn be written as is?
+                    match file_sink_write!("signed_poc_receipt_txn", receipt_sender, txn).await {
+                        Ok(_) => tracing::debug!("txn written!"),
+                        Err(_) => tracing::error!("unable to write txn"),
                     }
-                    Err(e) => tracing::error!("unable to process msg due to: {:?}", e),
+                } else {
+                    tracing::error!("unable to submit txn")
```

Comment:
good opportunity for @xandkar to swoop in and send this to a prometheus gauge?

Reply:
or else similar to how we were handling rewards submissions in the mobile rewards server for a while you could just change this up so `handle_submission_txn` returns a result instead of an option and then both `handle_txn_submission` and `file_sink_write!` are just using the `?` operator since the file sink write macro is already emitting debug and error traces on success and failure and that's the only thing we're currently doing in the above

Reply:
Updated 3667dd8